src/Type/Php/JsonThrowOnErrorDynamicReturnTypeExtension.php

```diff
@@ -51,8 +71,15 @@ public function getTypeFromFunctionCall(
 		Scope $scope,
 	): Type
 	{
+		// update type based on JSON_THROW_ON_ERROR
 		$argumentPosition = $this->argumentPositions[$functionReflection->getName()];
 		$defaultReturnType = ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
+
+		// narrow type for json_decode()
```

Comment:
nitpick: I think there's no need for this comment, this block reads quite well IMO and is good to understand
```suggestion
```